Upgrade to Ionic 6

robinzaal opened the pull request:
Oppgraderer Ionic til versjon 6.

- [x] Ny Datetimepicker-komponent
- [x] Icon
- [x] Input
- [x] Modal
- [x] Popover
- [x] Radio
- [x] Select
- [x] Textarea

robinzaal committed:
Add new datetime component, remove deprecated inputs from ion-datetime

robinzaal committed:
Update colors and add formatting options and defaults

robinzaal committed:
Get locale from user settings

robinzaal committed:
Replace all previous ion-datetime instances

robinzaal committed:
Merge branch 'develop' into upgrade/upgrade-to-ionic-6

robinzaal committed:
Update package-lock.json

robinzaal committed:
Update package-lock.json

robinzaal committed:
Merge remote-tracking branch 'origin/upgrade/upgrade-to-ionic-6' into upgrade/upgrade-to-ionic-6

robinzaal committed:
Update package-lock.json

robinzaal committed:
Left align button text by default, support text align input on component

robinzaal committed:
Make sure datetimepicker modal doesn't scroll on smaller screens

robinzaal committed:
Update modals to work with Ionic6

robinzaal committed:
Update modals to work with Ionic6

gruble commented on src/app/components/datetime-picker/datetime-picker.component.ts:
```diff
@@ -0,0 +1,66 @@
+import { firstValueFrom } from 'rxjs';
+import { getLangKeyString } from '../../modules/common-core/models/lang-key.enum';
+
+@Component({
```
Ønsker meg litt dokumentasjon her. Hva kan du bruke denne til? Hvilke input-parametre må du bruke? Kanskje noe om at localDate støtter toveis-binding?

robinzaal replied:
Komponenten har fått dokumentasjon nå.

gruble commented on src/app/components/datetime-picker/datetime-picker.component.ts:
```diff
@@ -0,0 +1,66 @@
+  styleUrls: ['./datetime-picker.component.scss']
+})
+export class DatetimePickerComponent implements OnInit {
+  @Input() localDate: string;
```
_kan_ det misforstås til at det bare kan være dato og ikke tid?
Hvilket format skal den ha?

robinzaal replied:
Endret navn og dokumentert bedre.

gruble reviewed (changes requested):
Jeg syntes det så bra ut, og den nye komponenten virka enkel å bruke.
Har noen kommentarer på manglende dokumentasjon og noen småting som bør fikses.

gruble commented on src/app/components/datetime-picker/datetime-picker.component.spec.ts:
```diff
@@ -0,0 +1,24 @@
+// import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
```
Hadde du tenkt å fikse denne testen? Funker ikke kjempebra nå i hvertfall;-)

robinzaal replied:
Fjernet test.

gruble commented on src/app/modules/registration/pages/snow/avalanche-activity/avalanche-activity-modal/avalanche-activity-modal.page.html:
```diff
@@ -1,59 +1,57 @@
-        doneText="{{'DIALOGS.OK' | translate}}" cancelText="{{'DIALOGS.CANCEL' | translate}}"
-        display-format="D. MMM, YYYY" [max]="maxDate" [(ngModel)]="startDate">
-      </ion-datetime>
+      <app-datetime-picker [(localDate)]='startDate' [maxDate]='maxDate'></app-datetime-picker>
```
Viser tidspunkt 00:00. Tror ikke tidspunkt er relevant her.
Sjekk selv under  Snø / Skredaktivitet /Dato

robinzaal replied:
La inn ny formattering her, og la til support for "format", slik at vi ikke trenger å vise tid ([https://ionicframework.com/docs/api/datetime#date-selection](https://ionicframework.com/docs/api/datetime#date-selection))

gruble commented on src/app/modules/registration/components/set-location-in-map/set-location-in-map.component.html:
```diff
@@ -1,131 +1,130 @@
+          <div class='flex-row'>
+            <div class='flex-grow box edit-button no-padding'>
+              <ion-icon name='calendar'></ion-icon>&nbsp;
+              <app-datetime-picker [(localDate)]='localDate' [maxDate]='maxDate'></app-datetime-picker>
```
I denne komponenten (observasjonspunkt- og tid) funker ikke lengre "nå-funksjonen".

robinzaal replied:
Ser ut til å fungere fint nå som vi fikset denne buggen i en annen PR.

gruble reviewed (changes requested):
Jeg opplevde en gang at hvis jeg går inn og redigerer et tidspunkt, får jeg ikke lov å scrolle timetallet til seinere enn det tidspunktet var. Mulig dette var en engangshendelse.
![ikke-flere-timer](https://user-images.githubusercontent.com/71138449/191266984-b985b5c2-7fd4-4b00-9e94-c11aa983aaa5.PNG)

robinzaal committed:
Merge remote-tracking branch 'origin/upgrade/upgrade-to-ionic-6' into upgrade/upgrade-to-ionic-6

# Conflicts:
#	package-lock.json

robinzaal committed:
Document code and make rename some properties to better reflect the functionality.

The component now supports different presentations.

Fix some issues from PR comments.

robinzaal committed:
Merge branch 'develop' into upgrade/upgrade-to-ionic-6

# Conflicts:
#	package-lock.json

gruble commented on src/app/components/datetime-picker/datetime-picker.component.ts:
```diff
@@ -11,15 +11,26 @@ import { getLangKeyString } from '../../modules/common-core/models/lang-key.enum
+  @Input() presentation: DatetimePresentation = 'date-time';
 
-  @Output() localDateChange = new EventEmitter<string>();
+  @Output() dateTimeChange = new EventEmitter<string>();
```
Flott med dokumentasjon! Litt pirkete her nå, men har fortsatt noen kommentarer:

- language mangler dok. Skriv gjerne om denne må brukes eller ikke og hva som er "default"
- dateTimeFormat mangler dok. Det sier seg kanskje sjøl, men siden dateTime er string, _kan_ det kanskje misforstås at det har noe med formatet på dateTime å gjøre? Dette gjelder vel bare hvordan datoen vises som tekst?
- dateTimeChange mangler dok

robinzaal replied:
Oppdatert.

gruble commented on src/app/components/datetime-picker/datetime-picker.component.ts:
```diff
@@ -0,0 +1,76 @@
+ * @param minDate - string
+ * @param maxDate - string
+ * @param dateTimeFormat - string
+ * @param textAlign - 'left' | 'center' | 'right'
```
Det er første gang jeg har sett bruk av @param-annoteringer i komponenter.
Jeg synes ikke @param-annoteringen tilføyer noe her, eller er det noe jeg ikke har skjønt?
I en service mener jeg det er nyttig, fordi da vises dokumentasjonen i "intellisense" (eller hva man kaller det) hvis vi prøver å bruke servicen i javascript-kode.
Jeg synes det er supert med en-linjes kommentarer som du har for hver input/output-parameter lengre ned-

robinzaal replied:
I WebStorm får man i alle fall opp informasjon om komponenten med denne typen annotering (se bilde under), så jeg tenker det er greit å la dette stå?
![image](https://user-images.githubusercontent.com/1227592/191690701-35f17bc4-0dab-4532-8251-9bb0c160aed0.png)

gruble replied:
OK!

gruble reviewed (changes requested):
Bra! Kun noen pirke-kommentarer på dokumentasjon igjen;-)

robinzaal committed:
Update documentation according to PR comments.

gruble reviewed (approved):
Bra jobba. Takk for tålmodigheten:)

gruble committed:
oppdatert package-lock med npm 6
